src/modules/mavlink/mavlink_main.cpp

```diff
@@ -479,7 +479,7 @@ Mavlink::forward_message(const mavlink_message_t *msg, Mavlink *self)
 		if (inst && (inst != self) && (inst->_forwarding_on)) {
 			// Pass message only if target component was seen before
 			if (inst->_receiver.component_was_seen(target_system_id, target_component_id)) {
-				inst->pass_message(msg);
+				inst->resend_message(msg);
```

Comment:
This is a bit questionable from a threading and locking perspective, each mavlink instance is a separate RX and TX thread. 

I think I understand the root cause though, let me think about an acceptable solution that doesn't intertwine separate mavlink instances and threads.